documentation/en/supraseal.md

```diff
@@ -0,0 +1,264 @@
+
+For calibnet
+```bash
+make batch-calibnet
+```
+
+NOTE: the build should be run on the target machine. Binaries won't be postable between CPU generations due to different
+AVX512 support.
+
+## Configuration
+
+* Run `curio calc batch-cpu` on the target machine to determine supported batch sizes for your CPU
+* Create a new layer configuration for the batch sealer, e.g. batch-machine1:
+
+```toml
+[Subsystems]
+EnableBatchSeal = true
+
+[Seal]
+LayerNVMEDevices = [
+  "0000:88:00.0",
+  "0000:86:00.0", 
+  # Add PCIe addresses for all NVMe devices to use
+]
+
+# Set to your desiced batch size (what the batch-cpu command says your CPU supports AND what you have nvme space for)
+BatchSealBatchSize = 32
+
+# pipelines can be either 1 or 2; 2 pipelines double storage requirements but in correctly balanced systems makes
+# layer hashing run 100% of the time, nearly doubling throughput
+BatchSealPipelines = 2
+
+# Set to true for Zen2 or older CPUs for compatibility
+SingleHasherPerThread = false
+```
+
+### Configure hugepages:
+
+This can be done by adding the following to `/etc/default/grub`. You need 36 1G hugepages for the batch sealer.
+
+```bash
+GRUB_CMDLINE_LINUX_DEFAULT="hugepages=36 default_hugepagesz=1G hugepagesz=1G"
+```
+
+Then run `sudo update-grub` and reboot the machine.
+
+Or at runtime:
+```bash
+sudo sysctl -w vm.nr_hugepages=36
+```
+
+Then check /proc/meminfo to verify the hugepages are available:
+```bash
+cat /proc/meminfo | grep Huge
+```
+
+Expect output like:
+```
+AnonHugePages:         0 kB
+ShmemHugePages:        0 kB
+FileHugePages:         0 kB
+HugePages_Total:      36
+HugePages_Free:       36
+HugePages_Rsvd:        0
+HugePages_Surp:        0
+Hugepagesize:    1048576 kB
+```
+
+Check that `HugePages_Free` is equal to 36, the kernel can sometimes use some of the hugepages for other purposes.
+
+### Setup NVMe devices for SPDK:
+
+NOTE: This is only needed while batch sealing is in beta, future versions of Curio will handle this automatically.
+
+```bash
+cd extern/supra_seal/deps/spdk-v22.09/
+env NRHUGE=36 ./scripts/setup.sh
+```
+
+### PC2 output storage
+
+Attach scratch space storage for PC2 output (batch sealer needs ~70GB per sector in batch - 32GiB for the sealed sector,
+and 36GiB for the cache directory with TreeC/TreeR and aux files)
+
+## Usage
+1. Start the Curio node with the batch sealer layer
+2. Add a batch of CC sectors:
+```bash
+curio seal start --now --cc --count 32 --actor f01234 --layers cluster --duration-days 365
+```
+
+3. Monitor progress - you should see a "Batch..." task running
+4. PC1 will take 3.5-5 hours, followed by PC2 on GPU
+5. After batch completion, the storage will be released for the next batch
+
+## Optimizing
+
+* Balance batch size, CPU cores, and NVMe drives to keep PC1 running constantly
+* Ensure sufficient GPU capacity to complete PC2 before next PC1 batch finishes
+* Monitor CPU, GPU and NVMe utilization to identify bottlenecks
+* Monitor hasher core utilisation
+
```

Comment:
* Ensure enough available space on regular drives for 32x (11*32GiB + 32GiB) sectors with layer files to sit 150 epochs, as well as 32 * 32 GiB for the complete sectors to offload to longer-term storage while the next batch arrives on the regular drives.

Reply:
this comment fits better below. My number is smaller than yours. Don't we only need 1 batch's layer files at a time, unless more batches complete in the 150 epochs?

Reply:
The storage requirements are already in the "Storage Recommendations" section. I've moved it up before setup for visibility

NI support "just" requires changes to porep challenge reading code, but on the Go side it's almost a no-op, changes are in the later stages of the pipeline.

Synthetic doesn't really make sense with supraseal machines, they just move too fast and have to have 2x space for sectors anyways